app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -243,26 +238,40 @@ class WifiDirectManager @Inject constructor(
     if (BuildConfig.DEBUG) {
       Log.d(TAG, "Starting handshake")
     }
-    peerGroupHandshakeAsyncTask = PeerGroupHandshakeAsyncTask(this)
-      .also { it.execute() }
+    var inetAddress: InetAddress?
```

Comment:
scope is too broad, declare it where it is assigned and it can be a val and then you will get the ebenfit of smart casting from the null check and can avoid the `!!`